doc/api/cli.md

```diff
@@ -156,6 +156,40 @@ Currently the support for run-time snapshot is experimental in that:
    a report in the [Node.js issue tracker][] and link to it in the
    [tracking issue for user-land snapshots][].
 
+### `--compile=path`
```

Comment:
perhaps the interface can be `node compile file.js > output`
similar to `node inspect`,
WDYT?

Reply:
@bnb when you mention "top level commands" are you referring to @MoLow's suggestion?

I'm also not sure about `--sea` being just an example to mention that an argument with more context would be beneficial here.

Reply:
@MoLow I did consider using `compile` like `deno compile` but I went ahead with `--compile` instead because of the reason @bnb stated.

> IMHO there should be two arguments. The first is a boolean --sea or --compile-binary that tells that we want to create a SEA. If only this argument is given, the output binary name will be the same as the input file. E.g. node --sea index.js would result in [./index] as a binary.
> 
> Then a second argument --compile-output or --compile-filename or --sea-filename that is optional and when you want to have a different name for the output binary.

@ovflowd looks good. We could keep adding more arguments for the other parts of sea, like the loader file for the vfs, baked cli args, etc.